docs/src/RunningOnGPUs.md

```diff
@@ -0,0 +1,52 @@
+# Running on GPUs
+
+ITensor provides package extensions for running tensor operations on a variety of GPU backends.
+You can activate a backend by loading the appropriate Julia GPU package alongside ITensors.jl
+and moving your tensors and/or tensor networks to an available GPU using that package's provided conversion functions.
+
+For example, you can load CUDA.jl to perform tensor operations on NVIDIA GPUs or Metal.jl to perform tensor operations on Apple GPUs:
+
+```julia
+using ITensors
+
+i, j, k = Index.((2, 2, 2))
+A = randomITensor(i, j)
+B = randomITensor(j, k)
+
+# Perform tensor operations on CPU
+A * B
+
+###########################################
+using CUDA # This will trigger the loading of `NDTensorsCUDAExt` in the background
+
+# Move tensors to NVIDIA GPU
+Acu = cu(A)
+Bcu = cu(B)
+
+# Perform tensor operations on NVIDIA GPU
+Acu * Bcu
+
+###########################################
+using Metal # This will trigger the loading of `NDTensorsMetalExt` in the background
+
+# Move tensors to Apple GPU
+Amtl = mtl(A)
+Bmtl = mtl(B)
+
+# Perform tensor operations on Apple GPU
+Amtl * Bmtl
+```
+
+ITensor currently provides
+package extensions for the following GPU backends:
+
+* [CUDA.jl](https://github.com/JuliaGPU/CUDA.jl) (NVIDIA GPUs)
+* [Metal.jl](https://github.com/JuliaGPU/Metal.jl) (Apple GPUs)
+* [AMDGPU.jl](https://github.com/JuliaGPU/AMDGPU.jl) (AMD GPUs)
+
+Our goal is to support all GPU backends which are supported by the [JuliaGPU organization](https://juliagpu.org).
+
+Some important caveats to keep in mind related to the ITensor GPU backends are:
+* only dense tensor operations are well supported right now. Block sparse operations (which arise when QN conservation is enabled) are under active development and either may not work or may be slower than their CPU counterparts,
+* certain GPU backends do not have native support for certain matrix decompositions like `svd`, `eigen`, and `qr` in which case we will perform those operations on CPU. If your calculation is dominated by those operations, there likely is no advantage to running it on GPU right now. CUDA generally has good support for native matrix decompositions, while Metal and AMD have more limited support right now, and
```

Comment:
What about an operation support table in this part? Do you guys have which operations are currently supported by each backend written somewhere I can copy from?

Reply:
I would prefer to keep it simpler and a bit more vague right now since it is still under development, that would be something we would then have to keep up to date which is extra work for us and easy to forget to do. Users can investigate those things themselves by looking at the documentation of the different backends. This is the kind of reason why we haven't documented it yet, since we are then documenting a moving target which takes away time from doing the work to improve the code, we are a small team and developer time is limited.

Reply:
I don't mean to come across as too negative on the idea of a support table, I do think it is a very good idea, I would just prefer to hold off on it for now and wait for our backends to become better tested and developed first.

Reply:
I understand your point and I'm fine with leaving the table out. Unsynchronized documentation tends to become more of a hassle than anything else.

Reply:
Thinking about it more, it may be easier to use a table rather than have this list here, so feel free to make a table and we can see how it looks/assess if it seems difficult to upkeep.

I think it could just be something like:
```
                                    CUDA     ROCm         Metal         oneAPI
Contractions (dense)  x              x                  x               N/A
QR (dense)                   x              On CPU      On CPU     N/A
SVD (dense)                x              On CPU      On CPU     N/A
Hermitian                     x              On CPU      On CPU    N/A
  eigendecomposition (dense)
Double precision         x              x                 N/A           N/A
Block sparse               N/A           N/A             N/A           N/A
```